### PR TITLE
MRG: Add ECoG plotting to plot_trans

### PR DESCRIPTION
**Pull request by Eric89GXL**
@choldgraf can you see if this works okay for your real data?

### CODE REVIEW

**Comment by choldgraf** [comment]
I'm having a hard time creating a montage file -> Info object using just ECoG channels.

Here's my code:

```python
mon = mne.channels.Montage(plt_pts, names[keep], 'ecog',
                           range(len(names[keep])))

keep_chans = list(names[keep])
n_chans = len(keep_chans)
info = mne.create_info(keep_chans, 100, 'ecog', mon)
```

and it's throwing this error about needing `eeg` channels:

```python
/Users/choldgraf/Dropbox/github/forks/python/mne-python/mne/channels/montage.py in _set_montage(info, montage, update_ch_names)
    658 
    659         if not _contains_ch_type(info, 'eeg'):
--> 660             raise ValueError('No EEG channels found.')
    661 
    662         sensors_found = []

ValueError: No EEG channels found.
```

When I just added an extra check for `ecog`, I then get a really weird error 

```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-10-8b5ec7b7bdb7> in <module>()
      4 keep_chans = list(names[keep])
      5 n_chans = len(keep_chans)
----> 6 info = mne.create_info(keep_chans, 100, 'ecog', mon)
      7 data = np.random.randn(n_chans, 10000)
      8 

/Users/choldgraf/Dropbox/github/forks/python/mne-python/mne/io/meas_info.py in create_info(ch_names, sfreq, ch_types, montage)
   1487                 raise TypeError('Montage must be an instance of Montage, '
   1488                                 'DigMontage, a list of montages, or filepath, '
-> 1489                                 'not %s.' % type(montage_))
   1490     info._check_consistency()
   1491     return info

TypeError: Montage must be an instance of Montage, DigMontage, a list of montages, or filepath, not <class 'mne.channels.montage.Montage'>.
```

I should note, the line at 1489 originally was `type(montage)` and I'm pretty sure it's supposed to be `type(montage_)`, just FYI.

I have no idea why it's deciding that a `Montage` is not of type `Montage`...

I also tried using `DigMontage` but I'm getting some weird errors because I don't have any information other than the 3d electrode positions (it seems to be assuming that I provide head shape etc)

Hope some of that helps...

**Comment by Eric89GXL** [comment]
See my example. Don't use `Montage`, use `DigMontage`.

**Comment by choldgraf** [comment]
```python
mon = mne.channels.DigMontage({'ecog': plt_pts})

keep_chans = list(names[keep])
n_chans = len(keep_chans)
info = mne.create_info(keep_chans, 100, 'ecog', mon)
```
Yields this error:
```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-12-ba2af6bbef0e> in <module>()
      5 keep_chans = list(names[keep])
      6 n_chans = len(keep_chans)
----> 7 info = mne.create_info(keep_chans, 100, 'ecog', mon)
      8 data = np.random.randn(n_chans, 10000)
      9 

/Users/choldgraf/Dropbox/github/forks/python/mne-python/mne/io/meas_info.py in create_info(ch_names, sfreq, ch_types, montage)
   1487                 raise TypeError('Montage must be an instance of Montage, '
   1488                                 'DigMontage, a list of montages, or filepath, '
-> 1489                                 'not %s.' % type(montage_))
   1490     info._check_consistency()
   1491     return info

TypeError: Montage must be an instance of Montage, DigMontage, a list of montages, or filepath, not <class 'mne.channels.montage.DigMontage'>.
```

super weird

**Comment by Eric89GXL** [comment]
See my example. Use `mon = mne.channels.DigMontage(dig_ch_pos=dig_ch_pos)`.

**Comment by choldgraf** [comment]
image

**Comment by Eric89GXL** [comment]
From Gitter chat with @choldgraf he mentioned that this works for his data, so this is ready for review/merge from my end.

@noreun feel free to try the `DigMontage` with `dig_ch_pos` to see if it works for you.

**Comment by choldgraf** [comment]
This did work for the test data in the example as well as my own dataset. The only comment I'd make on this one is ask why we're loading in a `.mat` file in the example. Why not lead an hdf5 file with h5io or something? Not a big deal tho.

Is there a place that explains the difference between `Montage` and `DigMontage`, and when to use one or the other?

**Comment by Eric89GXL** [comment]
> The only comment I'd make on this one is ask why we're loading in a .mat file in the example.

I want to make the MATLAB converts feel safe :)

> Is there a place that explains the difference between Montage and DigMontage, and when to use one or the other?

No, and we need it. From conversations with @teonbrooks I think the idea is that `Montage` is for standard, non-individualized electrode placement, whereas `DigMontage` is for individualized locations/digitizations of pretty much everything (fiducials, HPI coils, extra points, EEG electrodes, now ECoG electrodes). 

@teonbrooks do you have any time in the coming weeks to try to write a little tutorial or manual entry on this?

**Comment by choldgraf** [comment]
> I want to make the MATLAB converts feel safe :)

Fair enough :)

@teonbrooks if you guys throw together a tutorial of some kind, I can go through it and try to make sense of things. Since I'm pretty new to surface plotting in MNE, and have little background in MEG source plotting, I might be a decent test-case for whether it's a clear tutorial or not

**Comment by choldgraf** [comment]
Once this gets merged I can add a few lines showing how to take a snapshot of the brain view and plot scatterplots on top of it.

File: tutorials/plot_ecog.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+"""
+======================
+Working with ECoG data
+======================
```
Comment (agramfort) [inline tutorials/plot_ecog.py]: this page needs to be integrated in the tutorials page:

http://martinos.org/mne/stable/tutorials.html

**Comment by agramfort** [comment]
for the curious ones:

https://2987-1301584-gh.circle-artifacts.com/0/home/ubuntu/mne-python/doc/_build/html/auto_tutorials/plot_ecog.html

**Comment by Eric89GXL** [comment]
It is already

**Comment by kingjr** [comment]
Can you add some colors representing the colors on the example? Chris sent
me an example of that recently

**Comment by kingjr** [comment]
forget it it's outside the scope of this PR

File: mne/io/meas_info.py [hunk 2]
```diff
@@ -528,8 +528,12 @@ def _make_dig_points(nasion=None, lpa=None, rpa=None, hpi=None,
             raise ValueError(msg)
     if dig_ch_pos is not None:
         keys = sorted(dig_ch_pos.keys())
-        for key in keys:
-            dig.append({'r': dig_ch_pos[key], 'ident': int(key[-3:]),
+        try:
```
Comment (kingjr) [inline mne/io/meas_info.py]: can you add comments on this try except?

File: mne/viz/_3d.py [hunk 3]
```diff
@@ -308,6 +310,16 @@ def plot_trans(info, trans='auto', subject=None, subjects_dir=None,
         If True, plot the digitization points.
     ref_meg : bool
         If True (default False), include reference MEG sensors.
+    ecog_sensors : bool
```
Comment (kingjr) [inline mne/viz/_3d.py]: channels

File: mne/viz/_3d.py [hunk 4]
```diff
@@ -356,7 +368,17 @@ def plot_trans(info, trans='auto', subject=None, subjects_dir=None,
     if coord_frame not in valid_coords:
         raise ValueError('coord_frame must be one of %s' % (valid_coords,))
 
-    show_head = (subject is not None)
+    meg_picks = pick_types(info, meg=True, ref_meg=ref_meg)
+    eeg_picks = pick_types(info, meg=False, eeg=True, ref_meg=False)
+    ecog_picks = pick_types(info, meg=False, ecog=True, ref_meg=False)
+
+    if head is None:
+        if len(ecog_picks) == 0 and subject is not None:
```
Comment (kingjr) [inline mne/viz/_3d.py]: head = len(ecog_picks) == 0 and subject is not None

File: mne/viz/_3d.py [hunk 5]
```diff
@@ -388,29 +409,46 @@ def plot_trans(info, trans='auto', subject=None, subjects_dir=None,
...
-    if show_head:
+    if head:
         subjects_dir = get_subjects_dir(subjects_dir, raise_error=True)
         surfs['head'] = get_head_surf(subject, source=source,
                                       subjects_dir=subjects_dir)
     if 'helmet' in meg_sensors and len(meg_picks) > 0 and \
             (ch_type is None or ch_type == 'meg'):
         surfs['helmet'] = get_meg_helmet_surf(info, head_mri_t)
+    if brain is None:
+        if len(ecog_picks) > 0 and subject is not None:
```
Comment (kingjr) [inline mne/viz/_3d.py]: > 0 superfluous?

File: tutorials/plot_ecog.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+import mne
+from mne.viz import plot_trans
+
+print(__doc__)
+
+###############################################################################
+# Let's load some ECoG electrode locations and names, and turn them into
+# a :class:`mne.DigMontage` class.
+
+mat = loadmat(mne.datasets.misc.data_path() + '/ecog/sample_ecog.mat')
```
Comment (kingjr) [inline tutorials/plot_ecog.py]: is this a new dataset?

Comment (Eric89GXL) [inline tutorials/plot_ecog.py]: It's some electrodes from a real subject that I inaccurately morphed to `sample` for demonstration purposes, yes

File: mne/viz/_3d.py (hunk 5 above)
Comment (Eric89GXL) [inline mne/viz/_3d.py]: not strictly necessary but I find it reads more easily (explicit instead of implicit)

File: tutorials/plot_ecog.py (hunk 1 above)
Comment (Eric89GXL) [inline tutorials/plot_ecog.py]: It already is. You seem to be looking at / linking to the `stable` docs but look instead at the CircleCI build:

https://2987-1301584-gh.circle-artifacts.com/0/home/ubuntu/mne-python/doc/_build/html/tutorials.html

File: mne/viz/_3d.py (hunk 3 above)
Comment (Eric89GXL) [inline mne/viz/_3d.py]: Well we already have `meg_sensors` and `eeg_sensors`, so this is a bit more consistent. Do ECoG people never talk about sensors? Maybe they say "electrodes" but the consistency is a bit nice to have here, no?

**Comment by Eric89GXL** [comment]
Comments thus far addressed

File: mne/viz/_3d.py (hunk 3 above)
Comment (kingjr) [inline mne/viz/_3d.py]: ok. I thought everything was called channels in MNE.

**Comment by kingjr** [comment]
LGTM

**Comment by agramfort** [comment]
thx @Eric89GXL !

**Comment by teonbrooks** [comment]
> @teonbrooks do you have any time in the coming weeks to try to write a little tutorial or manual entry on this?

I will write up a little explainer for this. are you adding the channel location support or would you like me to do it?

**Comment by Eric89GXL** [comment]
It was already part of this PR, at least some of it

**Comment by mmagnuski** [comment]
@choldgraf 
just BTW - the issue with `isinstance` you had most likely comes from reloading modules (if anyone has good tips on how to reload modules without pain or strange artifacts - I'll be really thankful)

**Comment by choldgraf** [comment]
yeah I found that out the hard way @mmagnuski :) thanks for the tip...I still haven't found a good way to do this either, and it's become more of a problem now that I do more things interactively

**Pull request merged**